tracer/src/Datadog.Trace/Configuration/TracerSettings.cs

```diff
@@ -342,15 +338,30 @@ internal TracerSettings(IConfigurationSource? source, IConfigurationTelemetry te
             IsRunningInAzureAppService = config
                                         .WithKeys(ConfigurationKeys.AzureAppService.AzureAppServicesContextKey)
                                         .AsBool(false);
-            if (IsRunningInAzureAppService)
+
+            // we only want to instantiate ImmutableAzureAppServiceSettings if we are for sure running
+            // in AAS, or if we might be running in a Consumption Plan function.
+            var isMaybeAzureFunctionConsumptionPlan = config.WithKeys(ConfigurationKeys.AzureAppService.FunctionsExtensionVersionKey).AsString() is not null;
```

Comment:
We want to instantiate AzureAppServiceMetadata if we think we're running in a consumption plan function, which would have `IsRunningInAzureAppService=false` as it relies on a key set in an AAS specific location (which will never be populated in consumption plan functions)